schema/scope.go

```diff
@@ -93,16 +93,16 @@ func (s *ScopeSchema) Serialize(data any) (any, error) {
 	return s.RootObject().Serialize(data)
 }
 
-func (s *ScopeSchema) ApplyScope(externalScope Scope, namespace string) {
+func (s *ScopeSchema) ApplyNamespace(externalObjects map[string]*ObjectSchema, namespace string) {
 	// When the namespace is the default namespace, each scope should pass itself down.
-	var scopeToApply Scope
-	if namespace == DEFAULT_NAMESPACE {
-		scopeToApply = s
+	var objectsToApply map[string]*ObjectSchema
+	if namespace == SelfNamespace {
+		objectsToApply = s.Objects()
 	} else {
-		scopeToApply = externalScope
+		objectsToApply = externalObjects
 	}
```

Comment:
This code seems weird:  if the caller passes `SelfNamespace` as `namespace`, then we ignore what the caller passed in `externalObjects` and use some other value, instead.

Why can't/don't we just count on the caller to pass the correct objects list?  Since it has access to this method, the caller should have access to `s.Objects()`, right?

Reply:
The reason is because the scope passed in can be the "self" of a scope that calls this scope. That's because this scope can be a property in an object of another scope. So we need to always pass in `s`.

Reply:
The problem is the recursive case. So you have a scope in a scope in a scope. In that case, each one needs to apply its own objects. It doesn't need its parents' objects. In that case, `s.Objects()` != `externalObjects`

Reply:
Done.